File: benchmark-csharp/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;

namespace benchmark_csharp.Controllers;

[ApiController]
[Route("")]
public class HealthController : ControllerBase
{
    [HttpGet("ping")]
    public IActionResult Ping() => Content("OK", "text/plain");

    [HttpGet("error")]
    public IActionResult Error() => throw new Exception("Simulated exception");
}

File: benchmark-php/app/Http/Controllers/HealthController.php
<?php

namespace App\Http\Controllers;

use Symfony\Component\HttpKernel\Exception\HttpException;

class HealthController extends Controller
{
    public function ping()
    {
        return response("OK", 200)->header("Content-Type", "text/plain");
    }

    public function error()
    {
        throw new HttpException(500, 'Simulated exception');
    }
}
